x-pack/plugins/secops/server/lib/suricata/elasticsearch_suricata_adapter.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { camelCase, get, has } from 'lodash/fp';
+import { SuricataEvents } from '../../../common/graphql/types';
+import { FrameworkAdapter, FrameworkRequest } from '../framework';
+import { EventData, SuricataAdapter, SuricataRequestOptions } from './types';
+
+const EventFields = [
+  'suricata.eve.timestamp',
+  'suricata.eve.flow_id',
+  'suricata.eve.src_ip',
+  'suricata.eve.src_port',
+  'suricata.eve.event_type',
+  'suricata.eve.proto',
+  'suricata.eve.dest_ip',
+  'suricata.eve.dest_port',
+];
+
+export class ElasticsearchSuricataAdapter implements SuricataAdapter {
+  private framework: FrameworkAdapter;
+  constructor(framework: FrameworkAdapter) {
+    this.framework = framework;
+  }
+
+  public async getEvents(
+    request: FrameworkRequest,
+    options: SuricataRequestOptions
+  ): Promise<SuricataEvents[]> {
```

Comment:
In addition to the events, I think we need to return additional metadata about the response. For example, `elasticsearch` can return a partial set of results based on the availability of nodes in a cluster. We may need to convey this degraded state to the user.  It may make sense to see how [elasticsearch.js](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/about.html) models this.

Reply:
Yes agree, we should just create an issue for that. I do not think we can tackle this for the Friday deadline

Reply:
https://github.com/elastic/ingest-dev/issues/91